src/content/1.7/development/architecture/domain/services.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Domain services
+weight: 20
+---
+
+# Domain services
+
+As mentioned in [CQRS principles](cqrs.md), a single command is just a port, a representation of a single use case, therefore the command handler should not contain any domain logic by itself, but should be orchestrating other services to fulfill the use case instead.
+These services are responsible for logic such as:
+1. Persisting the entities (a.k.a ObjectModel in current prestashop architecture) data into database
+2. Validating the entities
+3. Performing various modifications of related data structures etc.
+
+Here are couple principles of the domain services: 
+1. It MUST use the existing `ObjectModel` for writes as long as those classes exist.
+2. If it uses any legacy service or object model it MUST stay in adapter namespace.
+3. If it uses any `ObjectModel` method related to data persistence, that method MUST be wrapped into a dedicated repository class which must ensure that no legacy exceptions are thrown (some reusable methods are present in [AbstractObjectModelRepository](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/AbstractObjectModelRepository.php))
+4. If `ObjectModel` is used for any write operation it MUST be validated by a dedicated validator class (some reusable methods are present in [AbstractObjectModelValidator](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/AbstractObjectModelValidator.php))
```

Comment:
Don't use `develop` for links, choose a branch or a tag, because the content might be deleted in a future commit.
```suggestion
4. If `ObjectModel` is used for any write operation it MUST be validated by a dedicated validator class (some reusable methods are present in [AbstractObjectModelValidator](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/Adapter/AbstractObjectModelValidator.php))
```